File: MlbDataServer/DataFetch/GameSummaryCreator.cs
using MlbDataServer.DataStructures;
using System;

namespace MlbDataServer.DataFetch
{
	public class GameSummaryCreator
	{
		private const string UrlBase = "http://gd2.mlb.com";

		private string BuildUrl(DateTime date)
		{
			var yearFolder = "year_" + date.ToString("yyyy");
			var monthFolder = "month_" + date.ToString("MM");
			var dayFolder = "day_" + date.ToString("dd");

			return UrlBase + "/components/game/mlb/" + yearFolder + "/" + monthFolder + "/" + dayFolder + "/master_scoreboard.xml";
		}

		public GameSummaryCollection GetSummaryCollection(DateTime date)
		{
			var url = BuildUrl(date);
			GameSummaryCollection gameSummaryCollection = null;

			var xmlLoader = new XmlLoader();
			gameSummaryCollection = xmlLoader.GetXml<GameSummaryCollection>(url);

			return gameSummaryCollection;
		}
	}
}

File: BaseballTheaterCore/ClientApp/MlbDataServer/game_summary_creator.ts
import {GameSummaryCollection} from "./Contracts";
import {Moment} from "moment/moment"
import {DataLoader} from "./Utils/data_loader";

export class GameSummaryCreator
{
	private static readonly urlBase = `http://gd2.mlb.com`;

	private static buildUrl(date: Moment)
	{
		const yearFolder = `year_${date.format("YYYY")}`;
		const monthFolder = `month_${date.format("MM")}`;
		const dayFolder = `day_${date.format("DD")}`;

		return `${this.urlBase}/components/game/mlb/${yearFolder}/${monthFolder}/${dayFolder}/master_scoreboard.xml`;
	}

	public static async getSummaryCollection(date: Moment)
	{
		const url = this.buildUrl(date);

		const gameSummaryCollectionData = await DataLoader.loadXml<GameSummaryCollection>(url, "gameSummaryCollection");

		const gameSummaryCollection = new GameSummaryCollection(gameSummaryCollectionData);

		return gameSummaryCollection;
	}
}
